src/test/scala/mesosphere/marathon/integration/TaskQueueIntegrationTest.scala

```diff
@@ -0,0 +1,24 @@
+package mesosphere.marathon.integration
+
+import mesosphere.marathon.integration.setup._
+import org.scalatest.{ GivenWhenThen, Matchers }
+
+class TaskQueueIntegrationTest extends IntegrationFunSuite with SingleMarathonIntegrationTest with GivenWhenThen with Matchers {
+  test("GET /v2/queue") {
+    Given("a new app that is not healthy")
+    val appId = testBasePath / "failing"
+    val app = v2AppProxy(appId, "v1", instances = 2, withHealth = true)
```

Comment:
That is not a safe way for an entry in the queue. Consider using constraints to make it impossible to launch:
```scala
val c = Protos.Constraint.newBuilder().setField("notExistent").setOperator(Operator.CLUSTER).setValue("na").build()
val app = V2AppDefinition("/app".toRootTestPath, constraints = Set(c), cmd = Some("na"), instances = 5, ports = Seq(new JInt(23)))
```